### PR TITLE
Enable Calcite ReduceExpressionsRule in order to simplify expressions

### PR DESCRIPTION
**Pull request by eolivelli**
With ReduceExpressionsRule Calcite tries to simplify more aggressively input expressions.
Stuff like: 
"SELECT from a where n = 1 and n is not null" becomes "SELECT from a where n = 1"
"SELECT from a where n < 1 and n < 2" becomes "SELECT from a where n < 1"

### CODE REVIEW

File: herddb-core/src/main/java/herddb/sql/CalcitePlanner.java [hunk 1]
```diff
@@ -505,6 +506,7 @@ private PlannerResult runPlanner(String defaultTableSpace, String query) throws
         RelDataType originalRowType = logicalPlan.getRowType();
         RelOptCluster cluster = logicalPlan.getCluster();
         final RelOptPlanner optPlanner = cluster.getPlanner();
+        optPlanner.addRule(ReduceExpressionsRule.FILTER_INSTANCE);
```
Comment (amitvc) [inline herddb-core/src/main/java/herddb/sql/CalcitePlanner.java]: @eolivelli what is the purpose of adding the Filter_Instance here?

Comment (eolivelli) [reply herddb-core/src/main/java/herddb/sql/CalcitePlanner.java]: This is the core of this change.
That Rule simplifies every expression, removing redundant filterd

Comment (amitvc) [reply herddb-core/src/main/java/herddb/sql/CalcitePlanner.java]: I see. I think this looks good to me.

**Comment by eolivelli** [comment]
One test is failing.
I have to fix it before merging

File: herddb-core/src/test/java/herddb/core/SimpleOperatorsTest.java [hunk 2]
```diff
@@ -294,28 +292,14 @@ public void simpleArithmeticOperationsTest() throws Exception {
             }
 
             // In expressions
-            // Warning: jsqlParser doesn't handle this kind of expressions in select clause
             try (DataScanner scan1 = scan(manager, "SELECT * FROM tblspace1.tsql WHERE '1' in (1,2,3)", Collections.emptyList())) {
```
Comment (eolivelli) [inline herddb-core/src/test/java/herddb/core/SimpleOperatorsTest.java]: @amitvc  this kind of query is not useful in real life.
We can change the behaviour

**Pull request merged**